tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/DebugWriter.java

```diff
@@ -0,0 +1,126 @@
+package org.unicode.icu.tool.cldrtoicu;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.logging.Formatter;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+
+import org.unicode.cldr.api.CldrDataSupplier;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Utility to allow any "mapper" class to trivially support a main method and useful
+ * logging behaviour to help avoid the need for ad-hoc logging via {@code System.out}.
+ *
+ * <p>In most cases a mapping class can just have a {@code main} method like:
+ * <pre>{@code
+ *   // Arguments: <output-dir> [<log-level>]
+ *   public static void main(String[] args) throws IOException {
+ *       DebugWriter.writeForDebugging(args, MapperClass::process);
+ *   }
+ * }</pre>
+ *
+ * <p>Note however that running this still requires that the {@code CLDR_DIR} system
+ * property is set.
+ */
+public final class DebugWriter {
+    private static final String PACKAGE_ROOT = "org.unicode.icu.tool.cldrtoicu";
+    private static final String PACKAGE_PREFIX = PACKAGE_ROOT + ".";
+
+    /**
+     * Writes the IcuData generated by the given function using the default {@code CLDR_DIR}
+     * system property.
+     *
+     * <p>This is a helper method to make it easy for each mapper to have its own main
+     * method for debugging, and it should not be used directly by {@code LdmlConverter}.
+     */
+    public static void writeForDebugging(String[] args, Function<CldrDataSupplier, IcuData> fn)
+            throws IOException {
+        writeMultipleForDebugging(args, src -> ImmutableList.of(fn.apply(src)));
+    }
+
+    /**
+     * Writes the IcuData generated by the given function using the default {@code CLDR_DIR}
+     * system property.
+     *
+     * <p>This is a helper method to make it easy for each mapper to have its own main
+     * method for debugging, and it should not be used directly by {@code LdmlConverter}.
+     */
+    public static void writeMultipleForDebugging(
+            String[] args, Function<CldrDataSupplier, Collection<IcuData>> fn)
+            throws IOException {
+        String cldrPath = System.getProperty("CLDR_DIR", System.getenv("CLDR_DIR"));
```

Comment:
Should this not call whatever CLDR code typically calls for reading properties/env vars?

Reply:
It can't really, that's not an available API. This reads a property and falls back to the environment variable, which is essentially the same behaviour.